bayes_opt/target_space.py

```diff
@@ -176,8 +181,12 @@ def register(self, params, target, constraint_value=None):
         """
         x = self._as_array(params)
         if x in self:
-            raise NotUniqueError('Data point {} is not unique'.format(x))
-
+            if self._allow_duplicate_points:
+                self.n_duplicate_points = self.n_duplicate_points + 1
+                warnings.warn(f'Data point {x} is not unique. {self.n_duplicate_points} duplicates registered.'
```

Comment:
FYI warnings are usually only shown once, so this will not give you status updates about the number of duplicates.

Reply:
I can't believe I had never noticed this!

Reply:
This will probably break the output of the `ScreenLogger`, right?
I think this way is perfectly fine, but if you want to use warnings & have them print repeatedly that would be possible by adding context, see [here](https://stackoverflow.com/a/22661838/12051450).

Reply:
As a sidenote regarding colours -- do you know if #208 has been fixed or not?

Reply:
yeah I guess it won't look great with screen logger.
perhaps a more elegant and appropriate way to do this is actually make the probing of duplicate points an event that can then get logged?

Reply:
At least for me I have no problems with the screen logger on windows so I closed #208.

Reply:
> perhaps a more elegant and appropriate way to do this is actually make the probing of duplicate points an event that can then get logged?

That makes sense. Still, I'm having trouble picturing how this would like in the context of the `ScreenLogger` output. Do you have a specific idea?

Reply:
Hey @till-m,
I thought about this a bit more.
Although the current approach does 'ruin' the output of the screen logger, I think that in most situations the screen logger should not be relied on anyway. I contemplated adding an extra column 'duplicate point (True/False)' but this seemed like overkill to address a situation that hopefully is not actually occurring that often anyway. 
Similarly, regarding making this an 'event' - to be honest this aspect of the code confuses me somewhat. From my perspective, seeing that the duplicate points are already clearly defined in the logs I'm not sure any more is needed...
Happy to hear other perspectives/approaches!

Reply:
I agree. Probably not worth worrying about. I would ship this as-is.